packages/web3torrent/src/contexts/omni-provider.tsx

```diff
@@ -0,0 +1,21 @@
+import {BudgetContext, useBudgetContext} from './budget-context';
+import {Web3TorrentClientContext, useWeb3TorrentClientContext} from './w3t-client-context';
+import React from 'react';
+import {ChannelContext, useChannelContext} from './channel-context';
+import {TorrentClientContext, useTorrentClientContext} from './torrent-context';
+
+export const OmniProvider: React.FC = ({children}) => {
+  const web3TorrentClientContext = useWeb3TorrentClientContext();
+  // TODO: Probably a more elegant way of doing this
+  return (
+    <Web3TorrentClientContext.Provider value={web3TorrentClientContext}>
+      <TorrentClientContext.Provider value={useTorrentClientContext({web3TorrentClientContext})}>
```

Comment:
To be honest, I am little surprised that `useState` and `useEffect` work inside a component property computation...